contracts/VetoGuard.sol

```diff
@@ -21,43 +22,26 @@ contract VetoGuard is
     IGnosisSafe public gnosisSafe;
     mapping(bytes32 => uint256) transactionQueuedBlock;
 
-    constructor(
-        address _owner,
-        uint256 _executionDelayBlocks,
-        address _vetoERC20Voting,
-        address _gnosisSafe
-    ) {
-        bytes memory initializeParams = abi.encode(
-            _executionDelayBlocks,
-            _owner,
-            _vetoERC20Voting,
-            _gnosisSafe
-        );
-        setUp(initializeParams);
-    }
-
     /// @notice Initialize function, will be triggered when a new proxy is deployed
     /// @param initializeParams Parameters of initialization encoded
     function setUp(bytes memory initializeParams) public override initializer {
         __Ownable_init();
         (
             uint256 _executionDelayBlocks,
             address _owner,
-            address _vetoERC20Voting,
-            address _gnosisSafe
-        ) = abi.decode(initializeParams, (uint256, address, address, address));
+            address _vetoERC20Voting
+        ) = abi.decode(initializeParams, (uint256, address, address));
 
         executionDelayBlocks = _executionDelayBlocks;
         transferOwnership(_owner);
         vetoERC20Voting = IVetoERC20Voting(_vetoERC20Voting);
-        gnosisSafe = IGnosisSafe(_gnosisSafe);
+        gnosisSafe = IGnosisSafe(msg.sender);
```

Comment:
Why are we assuming the Gnosis Safe is the msg.sender? That may not always be the case right?

Reply:
This is a good point - we may want to be a tad more granular about setting a safe. For example, someone may want to add the vetoguard - is there a way to make this less dependant on each other?

Reply:
Reviewed and solved this problem - If a user places address(0) then this means that the gnosis safe is the deployer. If not - then we should use the address provided by the user.

gnosisSafe = IGnosisSafe(
            _gnosisSafe == address(0) ? msg.sender : _gnosisSafe
        );

Reply:
Hmm I'm not sure a way around this contract need to know the Gnosis Safe address. It needs to be able to check on the Safe that signatures are valid before it allows them to be queued.